homework6.js

```diff
@@ -0,0 +1,212 @@
+// ---------
+// 1a)
+// ---------
+
+var Ctor = function() {
+  this.a = 0;
+  this.b = 1;
+}
+
+var obj1 = new Ctor();
+var obj2 = new Ctor();
+
+var obj3 = {};
+
+Ctor.call(obj3);
+
+obj1.c = 2;
+
+obj1.constructor.prototype.d = 3;
+
+// obj1 and obj2 will both have a property d equal to 3 because we set that
+// property on the prototype.
+// obj3 however, will not have a d property because we did not use the
+// new keyward when creating object 3, and so it does not share that prototype
+
+// ---------
+// 1b)
+// ---------
+
+// the difference is that for objA we overrode the prototype and set it equal
+// to a new object with properties of num and str. With objB we simply added
+// two properties to it's original prototype. So objA has a prototype of Object
+// and objB has a prototype of B (with an extra property constructor
+// referenceing the function that created it)
+
+// ---------
+// 2a and 2b)
+// ---------
+
+// Animal constructor
+var Animal = function(name) {
+  this.name = name;
+};
+
+Animal.prototype.move = function() { return 'walk' };
+Animal.prototype.layEgg = function() { return new Egg(Animal) };
+
+// Bird inherits from Animal
+var Bird = function(name) {
+  this.hasWings = true;
+  this.name = name;
+};
+
+Bird.prototype = new Animal();
+Bird.prototype.move = function() { return 'fly' };
+Bird.prototype.layEgg = function() { return new Egg(Bird) };
+
+// Fish inherits from Animal
+var Fish = function(name) {
+  this.hasGills = true;
+  this.name = name;
+};
+
+Fish.prototype = new Animal();
+Fish.prototype.move = function() { return 'swim' };
+Fish.prototype.layEgg = function() { return new Egg(Fish) };
+
+// Penguin inherits from Bird which inherits from Animal
+var Penguin = function(name) {
+  this.name = name;
+};
+
+Penguin.prototype = new Bird();
+Penguin.prototype.move = function() { return 'swim' };
+Penguin.prototype.layEgg = function() { return new Egg(Penguin) };
+
+// Egg constructor
+var Egg = function(parentAnimal) {
+  this.parentAnimal = parentAnimal;
+};
+
+Egg.prototype.hatch = function(name) { return new this.parentAnimal(name) };
+
+// ---------
+// 3a)
+// ---------
+
```

Comment:
3a is looking good. Inserts Page rank where appropriate and giving the right number of cards. Don't forget to add in functionality for the orientation of the card (specifically for tarot cards) indicated by a  boolean.